### PR TITLE
refactor: initial stab at a new initialize function.

### PR DESCRIPTION
**Pull request by davidjoy**
Now has explicit steps for configuring each service and then accepting (potentially) asynchronous handlers after configuration.  Only auth actually has a default handler.

### CODE REVIEW

File: src/base/handlers/auth.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { getAuthenticatedApiClient, getAuthenticatedUser, ensureAuthenticatedUser } from '../../auth';
```
Comment (abutterworth) [inline src/base/handlers/auth.js]: `getAuthenticatedApiClient` is now `getAuthenticatedHttpClient`

File: src/base/handlers/auth.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  if (requireAuthenticatedUser) {
+    authenticatedUser = await ensureAuthenticatedUser();
+  } else {
+    authenticatedUser = await getAuthenticatedUser();
+  }
+
+  if (authenticatedUser !== null && hydrateAuthenticatedUser) {
+    const { username } = authenticatedUser;
+    getAuthenticatedApiClient()
```
Comment (abutterworth) [inline src/base/handlers/auth.js]: Is it worth a comment here that this does not use await in order to allow the rest of the app bootstrapping to proceed? (I assume this is the reason)

File: src/base/initialize.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  authService = null,
+  i18nService = null,
+  requireAuthenticatedUser = false,
+  hydrateAuthenticatedUser = false,
+  messages,
+  handlers = {},
+} = {}) {
+  try {
+    configurePubSub(pubSubService);
+    await override(() => {}, handlers.pubSub);
```
Comment (abutterworth) [inline src/base/initialize.js]: This override function is a little strange to me at first glance. What dropping it and using defaults instead – something like:
```
const handlers = Object.create(defaultHandlers, _handlers);

// ...
configureConfig(configService, { pubSubService: getPubSubService() });
handlers.config();
publish(APP_CONFIG_INITIALIZED);
```

File: src/base/handlers/auth.js (hunk 2 above)
Comment (davidjoy) [reply src/base/handlers/auth.js]: Good call.

File: src/base/initialize.js (hunk 3 above)
Comment (davidjoy) [reply src/base/initialize.js]: Cool - I like that.

**Pull request merged**